bleak/backends/corebluetooth/scanner.py

```diff
@@ -113,6 +113,7 @@ async def get_discovered_devices(self) -> List[BLEDevice]:
                     details,
                     uuids=uuids,
                     manufacturer_data=manufacturer_data,
+                    delegate=details.manager().delegate(),
```

Comment:
It would probably be safer to use `self._manager.delegate()` since we have found that `details.manager()` is undocumented and may not always exist in the future.

Reply:
Yes I agree with that although `CentralManagerDelegate` is `self._manager.central_manager.delegate()` (checked `self._manager.delegate()` but  `self._manager` is not  the `CBCentralManager`. Pretty confusing name convention in my opinion. 😕

Reply:
Doh, I am probably to blame for the bad naming there. We can clean that up later.